boilerplate/app/app.tsx.ejs

```diff
@@ -1,83 +1,55 @@
-// Welcome to the main entry point of the app.
-//
-// In this file, we'll be kicking off our app or storybook.
-
+/**
+ * Welcome to the main entry point of the app. In this file, we'll
+ * be kicking off our app or storybook.
+ *
+ * Most of this file is boilerplate and you shouldn't need to modify
+ * it very often. But take some time to look through and understand
+ * what is going on here.
+ *
+ * The app navigation resides in ./app/navigation, so head over there
+ * if you're interested in adding screens and navigators.
+ */
 import "./i18n"
-import React, { useState, useEffect, useRef } from "react"
-import { YellowBox } from "react-native"
+import "./utils/ignore-warnings"
+import React, { useState, useEffect, useRef, FunctionComponent as Component } from "react"
 import { NavigationContainerRef } from "@react-navigation/native"
+import { SafeAreaProvider, initialWindowSafeAreaInsets } from "react-native-safe-area-context"
 <% if (props.useExpo) { -%>
   import { initFonts } from "./theme/fonts"
 <% } -%>
-import { contains } from "ramda"
-import { enableScreens } from "react-native-screens"
-import { SafeAreaProvider, initialWindowSafeAreaInsets } from "react-native-safe-area-context"
-
-import { RootNavigator, exitRoutes, setRootNavigation } from "./navigation"
-import { useBackButtonHandler } from "./navigation/use-back-button-handler"
-import { RootStore, RootStoreProvider, setupRootStore } from "./models/root-store"
 import * as storage from "./utils/storage"
-import getActiveRouteName from "./navigation/get-active-routename"
+import {
+  useBackButtonHandler,
+  RootNavigator,
+  canExit,
+  setRootNavigation,
+  useNavigationPersistence,
+} from "./navigation"
+import { RootStore, RootStoreProvider, setupRootStore } from "./models/root-store"
 
 // This puts screens in a native ViewController or Activity. If you want fully native
 // stack navigation, use `createNativeStackNavigator` in place of `createStackNavigator`:
 // https://github.com/kmagiera/react-native-screens#using-native-stack-navigator
+import { enableScreens } from 'react-native-screens';
 enableScreens()
 
-/**
- * Ignore some yellowbox warnings. Some of these are for deprecated functions
- * that we haven't gotten around to replacing yet.
- */
-YellowBox.ignoreWarnings([
-  "componentWillMount is deprecated",
-  "componentWillReceiveProps is deprecated",
-  "Require cycle:",
-])
-
-/**
- * Are we allowed to exit the app?  This is called when the back button
- * is pressed on android.
- *
- * @param routeName The currently active route name.
- */
-const canExit = (routeName: string) => contains(routeName, exitRoutes)
-
 export const NAVIGATION_PERSISTENCE_KEY = "NAVIGATION_STATE"
 
 /**
  * This is the root component of our app.
  */
-const App: React.FunctionComponent<{}> = () => {
+const App: Component<{}> = () => {
```

Comment:
Why the import of `FunctionComponent as Component` and then changing this to `Component`? Couldn't this introduce some possibility for confusion by users?

Reply:
I'm seeing this in quite a few locations. Even as the only change in some files. Is this a new convention i'm unaware of?

Reply:
I'm not a fan of how long `React.FunctionComponent` is when it doesn't really add much useful information to the developer. It's mainly there for TypeScript. So by making it shorter, it makes the component easier to read at a glance.